config/consumers.py

```diff
@@ -36,7 +35,7 @@ async def connect(self):
 
         for route in routes:
             for element in elements:
-                msg = await sync_to_async(partial(element.websocketmessage))
+                msg = await sync_to_async(lambda e: e.websocketmessage)(element)
```

Comment:
I'm not sure it matters, but curious why we're changing this?

Reply:
TranslatorConsumer was barfing hundreds of logs faster than i could hit ctrl-c about this. I'm not gonna lie I don't 100% understand why it started happening all of a sudden (was on a fresh branch with no code from myself added), but this fixes it. I think something nested in there wasn't being called asynchronously? This was the error that was being fired off at an insanely fast rate if I added an entry.

```python
django-1  | INFO:     connection closed
django-1  | INFO 2024-12-24 16:36:02,399 server 48 274929645376 connection closed
django-1  | INFO 2024-12-24 16:36:02,403 consumers 48 274929645376 Translator connected
django-1  | INFO:     ('172.18.0.7', 34334) - "WebSocket /ws/route_manager/translator_block/" [accepted]
django-1  | INFO 2024-12-24 16:36:02,406 websockets_impl 48 274929645376 ('172.18.0.7', 34334) - "WebSocket /ws/route_manager/translator_block/" [accepted]
django-1  | INFO:     connection open
django-1  | INFO 2024-12-24 16:36:02,409 server 48 274929645376 connection open
django-1  | ERROR:    Exception in ASGI application
django-1  | Traceback (most recent call last):
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/db/models/fields/related_descriptors.py", line 218, in __get__
django-1  |     rel_obj = self.field.get_cached_value(instance)
django-1  |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/db/models/fields/mixins.py", line 15, in get_cached_value
django-1  |     return instance._state.fields_cache[cache_name]
django-1  |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^
django-1  | KeyError: 'websocketmessage'
django-1  | 
django-1  | During handling of the above exception, another exception occurred:
django-1  | 
django-1  | Traceback (most recent call last):
django-1  |   File "/usr/local/lib/python3.12/site-packages/uvicorn/protocols/websockets/websockets_impl.py", line 184, in run_asgi
django-1  |     result = await self.app(self.scope, self.asgi_receive, self.asgi_send)
django-1  |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/uvicorn/middleware/proxy_headers.py", line 75, in __call__
django-1  |     return await self.app(scope, receive, send)
django-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/channels/routing.py", line 71, in __call__
django-1  |     return await application(scope, receive, send)
django-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/channels/routing.py", line 150, in __call__
django-1  |     return await application(
django-1  |            ^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/channels/consumer.py", line 94, in app
django-1  |     return await consumer(scope, receive, send)
django-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/channels/consumer.py", line 58, in __call__
django-1  |     await await_many_dispatch(
django-1  |   File "/usr/local/lib/python3.12/site-packages/channels/utils.py", line 51, in await_many_dispatch
django-1  |     await dispatch(result)
django-1  |   File "/usr/local/lib/python3.12/site-packages/channels/consumer.py", line 73, in dispatch
django-1  |     await handler(message)
django-1  |   File "/usr/local/lib/python3.12/site-packages/channels/generic/websocket.py", line 173, in websocket_connect
django-1  |     await self.connect()
django-1  |   File "/app/config/consumers.py", line 38, in connect
django-1  |     msg = await sync_to_async(element.websocketmessage)()
django-1  |                               ^^^^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/db/models/fields/related_descriptors.py", line 236, in __get__
django-1  |     rel_obj = self.get_object(instance)
django-1  |               ^^^^^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/db/models/fields/related_descriptors.py", line 199, in get_object
django-1  |     return qs.get(self.field.get_reverse_related_filter(instance))
django-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/db/models/query.py", line 633, in get
django-1  |     num = len(clone)
django-1  |           ^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/db/models/query.py", line 380, in __len__
django-1  |     self._fetch_all()
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/db/models/query.py", line 1881, in _fetch_all
django-1  |     self._result_cache = list(self._iterable_class(self))
django-1  |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/db/models/query.py", line 91, in __iter__
django-1  |     results = compiler.execute_sql(
django-1  |               ^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/db/models/sql/compiler.py", line 1560, in execute_sql
django-1  |     cursor = self.connection.cursor()
django-1  |              ^^^^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/utils/asyncio.py", line 24, in inner
django-1  |     raise SynchronousOnlyOperation(message)
django-1  | django.core.exceptions.SynchronousOnlyOperation: You cannot call this from an async context - use a thread or sync_to_async.
django-1  | ERROR 2024-12-24 16:36:02,421 websockets_impl 48 274929645376 Exception in ASGI application
django-1  | Traceback (most recent call last):
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/db/models/fields/related_descriptors.py", line 218, in __get__
django-1  |     rel_obj = self.field.get_cached_value(instance)
django-1  |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/db/models/fields/mixins.py", line 15, in get_cached_value
django-1  |     return instance._state.fields_cache[cache_name]
django-1  |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^
django-1  | KeyError: 'websocketmessage'
django-1  | 
django-1  | During handling of the above exception, another exception occurred:
django-1  | 
django-1  | Traceback (most recent call last):
django-1  |   File "/usr/local/lib/python3.12/site-packages/uvicorn/protocols/websockets/websockets_impl.py", line 184, in run_asgi
django-1  |     result = await self.app(self.scope, self.asgi_receive, self.asgi_send)
django-1  |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/uvicorn/middleware/proxy_headers.py", line 75, in __call__
django-1  |     return await self.app(scope, receive, send)
django-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/channels/routing.py", line 71, in __call__
django-1  |     return await application(scope, receive, send)
django-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/channels/routing.py", line 150, in __call__
django-1  |     return await application(
django-1  |            ^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/channels/consumer.py", line 94, in app
django-1  |     return await consumer(scope, receive, send)
django-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/channels/consumer.py", line 58, in __call__
django-1  |     await await_many_dispatch(
django-1  |   File "/usr/local/lib/python3.12/site-packages/channels/utils.py", line 51, in await_many_dispatch
django-1  |     await dispatch(result)
django-1  |   File "/usr/local/lib/python3.12/site-packages/channels/consumer.py", line 73, in dispatch
django-1  |     await handler(message)
django-1  |   File "/usr/local/lib/python3.12/site-packages/channels/generic/websocket.py", line 173, in websocket_connect
django-1  |     await self.connect()
django-1  |   File "/app/config/consumers.py", line 38, in connect
django-1  |     msg = await sync_to_async(element.websocketmessage)()
django-1  |                               ^^^^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/db/models/fields/related_descriptors.py", line 236, in __get__
django-1  |     rel_obj = self.get_object(instance)
django-1  |               ^^^^^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/db/models/fields/related_descriptors.py", line 199, in get_object
django-1  |     return qs.get(self.field.get_reverse_related_filter(instance))
django-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/db/models/query.py", line 633, in get
django-1  |     num = len(clone)
django-1  |           ^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/db/models/query.py", line 380, in __len__
django-1  |     self._fetch_all()
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/db/models/query.py", line 1881, in _fetch_all
django-1  |     self._result_cache = list(self._iterable_class(self))
django-1  |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/db/models/query.py", line 91, in __iter__
django-1  |     results = compiler.execute_sql(
django-1  |               ^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/db/models/sql/compiler.py", line 1560, in execute_sql
django-1  |     cursor = self.connection.cursor()
django-1  |              ^^^^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/utils/asyncio.py", line 24, in inner
django-1  |     raise SynchronousOnlyOperation(message)
django-1  | django.core.exceptions.SynchronousOnlyOperation: You cannot call this from an async context - use a thread or sync_to_async.
django-1  | ERROR:    closing handshake failed
```

Reply:
Yuck, ok, thanks.